File: TodoApp.Client/src/interfaces/Requests/TodoItemRequest.ts
import type dayjs from "dayjs";
import type { Tier } from "../../commons";

export default interface TodoItemRequest {
  id?: string;
  title: string;
  description: string;
  dueDate: dayjs.Dayjs;
  isCompleted: boolean;
  priority: Tier;
  completedOn: dayjs.Dayjs;
  todoListId: string;
}

File: TodoApp.Server/src/Todo.DTOs/Requests/TodoItemRequest.cs
using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Todo.Domain.Enums;

namespace Todo.DTOs.Requests
{
    public class TodoItemRequest
    {
        public Guid Id { get; set; }

        [Required(ErrorMessage = "Title is required")]
        [StringLength(200, ErrorMessage = "Title must not exceed 200 characters")]
        public string Title { get; set; }

        [StringLength(500, ErrorMessage = "Description must not exceed 500 characters")]
        public string Description { get; set; }
        public DateTime DueDate { get; set; }
        public bool IsCompleted { get; set; }
        public Tier Priority { get; set; }
        public DateTime? CompletedOn { get; set; }
        public Guid? TodoListId { get; set; }
    }
}
